Sources/Datadog/Core/System/MobileDevice.swift

```diff
@@ -111,33 +111,41 @@ internal class MobileDevice {
 /// Observes "Low Power Mode" setting changes and provides `isLowPowerModeEnabled` value in a thread-safe manner.
 ///
 /// Note: this was added in https://github.com/DataDog/dd-sdk-ios/issues/609 where `ProcessInfo.isLowPowerModeEnabled` was considered
-/// not thread-safe on iOS 15. With this monitor, we change from pulling to push model for reading this property. Now, it will never be read simultaneously
-/// by multiple SDK threads - instead it will be read only once after LPM setting change and bridged to other threads through thread-safe `ValuePublisher`.
-///
-/// This should mitigate the crash originating in our SDK. We can't however prevent other code (e.g. application code) from reading this value simultaneously
-/// and causing a deadlock with SDK reads - ref. radar raised with Apple: FB9661108.
+/// not thread-safe on iOS 15. We suspect a bug present in iOS 15, where accessing `processInfo.isLowPowerModeEnabled` within a pending
+/// `.NSProcessInfoPowerStateDidChange` completion handler can sometimes lead to `_os_unfair_lock_recursive_abort` crash. The issue
+/// was reported to Apple, ref.: https://openradar.appspot.com/FB9741207
 private final class LowPowerModeMonitor {
     var isLowPowerModeEnabled: Bool {
         publisher.currentValue
     }
 
     private let publisher: ValuePublisher<Bool>
     private let notificationCenter: NotificationCenter
+    private let notificationQueue: OperationQueue
```

Comment:
Since it uses the `.main` queue, do we have to keep it has a property?

Reply:
No reason, updated for minimal code 👍